src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -51,6 +51,10 @@ The .NET Foundation licenses this file to you under the MIT license.
   <PropertyGroup Condition="'$(SuppressAotAnalysisWarnings)' == 'true'">
     <EnableAotAnalyzer Condition="'$(EnableAotAnalyzer)' == ''">false</EnableAotAnalyzer>
   </PropertyGroup>
+  
+  <PropertyGroup Condition="'$(SuppressSingleFileAnalysisWarnings)' == 'true'">
```

Comment:
Placing this in this file means that SuppressSingleFileAnalysisWarnings will only disable the single file analyzer for projects that do PublishAot. We'd ideally also respect this for PublishSingleFile.

I wonder if we should interpret these SuppressXXXWarnings in the Roslyn analyzer itself instead of turning SuppressXXXWarnings into EnableXXXAnalyzer in various targets (currently SuppressAot disables the analyzer in AOT targets, SuppressTrim disables the analyzer in ILLinker targets, and SuppressSingleFile doesn't really have suitable targets).

@sbomer @vitek-karas for thoughts.

Reply:
I definitely agree it should apply no matter which way single-file analysis is triggered. Technically doing this in MSBuild is a bit "Faster" as it avoid loading the analyzer completely in some cases, but it adds complexity for us. Doing this in the analyzer does feel like a simpler solution, but I'll leave that up to Sven as he knows the ins and outs of the MSBuilds parts of this much better.